packages/components/chip/src/useChipElement.tsx

```diff
@@ -0,0 +1,67 @@
+import { Slot } from '@spark-ui/slot'
+import { useCombinedState } from '@spark-ui/use-combined-state'
+import React from 'react'
+
+export const useChipElement = ({
+  onClick,
+  asChild,
+  pressed,
+  defaultPressed,
+  disabled,
+}: {
+  onClick?: (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    args: { pressed: boolean }
+  ) => void
+  asChild?: boolean
+  pressed?: boolean
+  defaultPressed?: boolean
+  disabled?: boolean
+}): {
+  Element:
+    | React.ForwardRefExoticComponent<
+        React.HTMLAttributes<HTMLElement> & {
+          children?: React.ReactNode
+        } & React.RefAttributes<HTMLElement>
+      >
+    | React.ElementType
+  chipProps:
+    | {
+        type: 'button'
+        'aria-pressed'?: boolean
+        'data-state'?: 'on' | 'off'
+        onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
+        disabled?: true
+      }
+    | {
+        'aria-disabled'?: true
+      }
```

Comment:
To improve legibility, maybe we could extract that type into its own variable., e.g.:
```tsx
interface ReturnedValue {
  Element: ...,
  chipProps: ...
}
```

Reply:
or 

```
type ChipComputedProps = {
        type: 'button'
        'aria-pressed'?: boolean
        'data-state'?: 'on' | 'off'
        onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
        disabled?: true
      }
    | {
        'aria-disabled'?: true
      }
```

then use this type in the interface (I just like the idea of making it explicit that the hook is computing the prop for the chip ^^") but its pretty similar
```
interface ChipContextValue {
  Element: 
}
```